replica/database.cc

```diff
@@ -2344,15 +2344,20 @@ future<> database::flush_keyspace_on_all_shards(sharded<database>& sharded_db, s
     });
 }
 
-future<> database::snapshot_tables_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, std::vector<sstring> table_names, sstring tag, bool skip_flush) {
-    co_await coroutine::parallel_for_each(table_names, [&, ks_name, tag = std::move(tag), skip_flush] (const auto& table_name) -> future<> {
+future<> database::snapshot_table_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, sstring table_name, sstring tag, bool skip_flush) {
+    // FIXME: indentation
         if (!skip_flush) {
             co_await flush_table_on_all_shards(sharded_db, ks_name, table_name);
         }
         co_await sharded_db.invoke_on_all([ks_name, &table_name, tag, skip_flush] (replica::database& db) {
             auto& t = db.find_column_family(ks_name, table_name);
             return t.snapshot(db, tag);
         });
+}
+
+future<> database::snapshot_tables_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, std::vector<sstring> table_names, sstring tag, bool skip_flush) {
+    co_await coroutine::parallel_for_each(table_names, [&, ks_name, tag = std::move(tag), skip_flush] (const auto& table_name) -> future<> {
```

Comment:
I'll see if legacy parallel_for_each can be used here without need for extra allocation.

Reply:
I think it's okay to use coroutines for administrative code, since it's likely to gain more code and emphasis should be on safety and readability, not performance.